c_examples/mem-dump.c

```diff
@@ -0,0 +1,47 @@
+#include "libmicrovmi.h"
+#include <stdio.h>
+#include <string.h>
+
+size_t PAGE_SIZE = 4096;
+
+void dump_memory(MicrovmiContext* driver, const char* vm_name) {
+    if (microvmi_pause(driver) == MicrovmiSuccess) {
+        printf("Paused.\n");
+    } else {
+        printf("Unable to pause VM.\n");
+        return;
+    }
+    uint64_t max_address;
+    if (microvmi_get_max_physical_addr(driver, &max_address) == MicrovmiSuccess) {
```

Comment:
Curiously, I couldn't run this example.
I always had a max address of `0`:
![Capture d’écran de 2020-01-19 22-47-43](https://user-images.githubusercontent.com/964610/72689086-ee7cee80-3b0d-11ea-928a-432a6a408502.png)

Reply:
Did you initialize the driver with DriverType = None or 0? Because this is equal to initalizing the driver with `DriverType::Dummy` since enums are just ints in C and 0 is the first enum value.

Reply:
Gotta figure out a way to make this parameter optional...